packages/gestalt-datepicker/src/DateRange/InternalDatePicker.tsx

```diff
@@ -95,9 +141,17 @@ const InternalDatePickerWithForwardRef = forwardRef<HTMLInputElement, ModifiedPr
             previousMonthButtonLabel={
               <Icon accessibilityLabel="" color="default" icon="arrow-back" size={16} />
             }
-            selected={rangeStartDate ?? undefined}
+            selected={
+              selectedRange === DateRangeType.Primary
```

Comment:
Why do we need this now, and we didn't have it earlier?

Reply:
This is because the logic for supporting both ranges (primary and secondary) was handled in the `DateRange.tsx` component itself, but now we also need to know which one is selected inside the InternalDatePicker in order to apply the correct styles to the "inactive range". Previously this component only had access to the "active" range.